### PR TITLE
DW initial revisions

### PR DESCRIPTION
**Pull request by dominikwelke**
hi @CarinaFo 

here my initial revisions that make it work (i think).
main changes:
- the `nanmean` function had to be swapped somewhere else, 
- applying your method is now only allowed if data are preloaded (which makes things easier and i think is valid).
- i added an initial test

it's just quick and dirty fixes.. e.g., the test could be much better.
also not sure if we should really change `mean` to `nanmean` in the global util function, or whether it makes more sense to allow `nanmean` as an additional separate option and call it based on a (to be implemented) flag, that epochs were rejected (`_check_combine` is only used in `epochs.py` and `tfr.py`, probably for a similar case, but who knows..).

### CODE REVIEW

File: mne/epochs.py [hunk 1]
```diff
@@ -1209,7 +1212,7 @@ def _compute_aggregate(self, picks, mode="mean"):
                 n_events += 1
 
             if n_events > 0:
-                data = np.nanmean(data)
+                data /= n_events
```
Comment (CarinaFo) [inline mne/epochs.py:1215]: I am still a bit confused about this part of the function, to be honest, maybe a comment would be good at that point

**Review by CarinaFo — APPROVED** [review]
I think I would add a comment to the compute_aggregate function, it's rather confusing what is actually happening there. The test looks good, but again a few comments would be nice and help the reviewer :)

**Comment by CarinaFo** [comment]
comments to the compute_aggregate function and the test function

**Review by CarinaFo — APPROVED** [review]

File: mne/epochs.py (hunk 1 above)
Comment (dominikwelke) [reply mne/epochs.py:1215]: this whole stuff only happens in the case of not-preloaded data, and thats why i guess it *has* to be somehow hacky and hard to follow ;) 
in this case there is no numpy array of data in memory that we can simply access and change (probably only a view or so..)
so they create this mock `data` object, fill it up by some loop through the epochs object and then average that.

but that's also why it's not really relevant for us - i doubt that we could even set not-preloaded data to `nan` easily without larger changes (i assume this info would have to be stored in some meta data and then applied when finally loading at some later point?), so it doenst make sense to implement an averaging function for this case.

Comment (dominikwelke) [reply mne/epochs.py:1215]: not super sure about the second part of my reply, maybe it is easier than i think. still, if we ask data to be preloaded, it doesnt apply

**Comment by dominikwelke** [comment]
> I think I would add a comment to the compute_aggregate function, it's rather confusing what is actually happening there. The test looks good, but again a few comments would be nice and help the reviewer :)

it's a private function, so this is why there isnt any. but i agree that a few hints would help ppl like us :)

File: mne/epochs.py (hunk 1 above)
Comment (CarinaFo) [reply mne/epochs.py:1215]: okey, thank you for the explanation, now I finally understand what is going on.
